tools/clang/test/CodeGenHLSL/literals_Mod.hlsl

```diff
@@ -60,6 +62,9 @@ float test() :Foo {
   VERIFY_TYPES(uint, overload2(2u, 2));
   VERIFY_TYPES(uint, overload2(2, 2ul));
   VERIFY_TYPES(uint, overload2(2ul, 2));
+  // TODO : half is converted to float. Must have overload function with half parameters when true half type is supported.
+  VERIFY_TYPES(half, overload2(1.0h, 1.0));
+  VERIFY_TYPES(half, overload2(1.0, 1.0H));
```

Comment:
overload2 has no half overload, so if half were supported, this would fail.  Further, we should promote half to float and not the other way around, so the return type expected would be float, even if half was supported and there was a half overload.